plugins/inputs/firehose/firehose_request.go

```diff
@@ -0,0 +1,156 @@
+package firehose
+
+import (
+	"compress/gzip"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf/config"
+)
+
+type firehoseRecord struct {
+	EncodedData string `json:"data"`
+}
+
+type firehoseRequestBody struct {
+	RequestID string           `json:"requestId"`
+	Timestamp int64            `json:"timestamp"`
+	Records   []firehoseRecord `json:"records"`
+}
+
+// https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html#requestformat
+type firehoseRequest struct {
+	req                *http.Request
+	body               firehoseRequestBody
+	responseStatusCode int
+}
+
+func (r *firehoseRequest) authenticate(expectedAccessKey config.Secret) error {
+	if expectedAccessKey.Empty() {
+		return nil
+	}
```

Comment:
If the expected key is empty, it means that there is no authentication configured on telegraf and thus we don't expect any key. Therefore, even if a key is provided in the incoming data it doesn't matter

Reply:
Please reflect this in the `sample.conf` comment and also make this very clear in the README! This is somewhat unexpected to me as it means "accept all" which is a really bad security strategy. ;-)

Reply:
I had kept authentication optional so that if the `access_key` is not set in the configuration block it means accept all. Otherwise, we’d check the incoming request for access key header and compare it with the one in the configuration block. However, we can also update the plugin to make the access key as something required and not optional. What do you suggest?